src/Workspaces/Core/Portable/Workspace/Solution/LoadableTextAndVersionSource.cs

```diff
@@ -31,6 +46,67 @@ private Task<TextAndVersion> LoadAsync(CancellationToken cancellationToken)
 
         private TextAndVersion LoadSynchronously(CancellationToken cancellationToken)
             => Source.Loader.LoadTextSynchronously(Options, cancellationToken);
+
+        public bool TryGetValue([MaybeNullWhen(false)] out TextAndVersion value)
+        {
+            value = _instance;
+            if (value != null)
+                return true;
+
+            return _weakInstance != null && _weakInstance.TryGetTarget(out value) && value != null;
```

Comment:
It may be worth adding a comment that once _weakInstance is non-null, it'll never be null again. As this is absolutely depending on that behavior, unlike the earlier read from _instance. Although I think that follows the same rule, so it's not clear to me why we're using a local in one place but not the other.

Reply:
sure.

Reply:
> It may be worth adding a comment that once _weakInstance is non-null, it'll never be null again.

Done.